wordpress-mobile/WordPress-Android
# Fix Unsupported Block Editor on Accounts Created Prior to December 2018

Fixes the Android side of wordpress-mobile/gutenberg-mobile#3425

`gutenberg`: https://github.com/WordPress/gutenberg/pull/32395
`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3573

## Description

The `/sites/{site_id}/gutenberg` endpoint is [unreliable](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3425#issuecomment-840223555) but was being used to check if Gutenberg was available on a site's backend, and if so, disable the Unsupported Block Editor (UBE).

Since the majority of users will have Gutenberg enabled, it makes sense to remove usage of this unreliable API endpoint. That's done by [removing the check for the Gutenberg editor](https://github.com/wordpress-mobile/WordPress-Android/pull/14762/files#diff-03bdc0871f050216e3aa83ee21fede80b81854a6e68b1694652f78f42653639f) in this PR.

To account for the removal, the loading and failed screens are updated in [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/32395).

## Testing

To verify that the UBE now loads for Jetpack sites on an account created prior to December 2018:

* Sign in using an account created before Dec 2018.
* Locate a post with an unsupported block on an Atomic site.
* Open the post and tap the (?) to open the web editor (UBE).
* Ensure the UBE opens as expected and you can successfully edit the unsupported block.

The UBE test cases outlined [here](https://github.com/wordpress-mobile/test-cases/blob/dc78dcff1c7d01a3c6f10a023b053ad4566d3ba1/test-cases/gutenberg/unsupported-block-editing.md) were used as a guide for further testing:

```
- [ ] TC001 - User can edit unsupported blocks on Simple WP.com sites
- [ ] TC002 - User can discard edits to an unsupported blocks on Simple WP.com sites
- [ ] TC003 - Editing unsupported blocks is allowed on Gutenberg-enabled Atomic sites
- [ ] TC004 - Editing unsupported blocks is disallowed on Classic-enabled Atomic sites
- [ ] TC005 - Editing unsupported blocks is enabled on self-hosted sites accessed via Jetpack
- [ ] TC006 – Editing unsupported blocks is disallowed on self-hosted sites accessed via their own username and password
```

When going through each test case, verify that the following is true:

* The message warning users that Gutenberg is required is displayed when the UBE is loading.
* When testing `TC004`, the failed screen is displayed to users after the progress screen has been visible for 10 seconds. 

## Regression Notes
1. Potential unintended areas of impact

This could somehow cause issues with the Unsupported Block Editor loading, so it's important to test this across all site types.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing was carried out, following the steps outlined in the Gutenberg PR.

3. What automated tests I added (or what prevented me from doing so)

This PR adds a loading screen that relies on the underlying network request, which seems tricky to account for using automated tests.

PR submission checklist:
